backend/custom_models.py

```
@@ -0,0 +1,165 @@
+    Raises:
+        ValueError: If model already exists or inputs are invalid
+    """
+    hf_repo_id = hf_repo_id.strip()
+    display_name = display_name.strip()
+    
+    if not hf_repo_id:
+        raise ValueError("HuggingFace repo ID is required")
+    if not display_name:
+        raise ValueError("Display name is required")
+    if "/" not in hf_repo_id:
+        raise ValueError("HuggingFace repo ID must be in format 'owner/model-name'")
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)